File: backend/app/apis/system/personnel.py
from app.corelibs.custom_router import APIRouter
from app.utils.response import HttpResponse

from app.schemas.system.personnel import (
    PersonnelIn, PersonnelUpdate, PersonnelDel, PersonnelQuery,
    PersonnelQualificationCertificateIn, PersonnelQualificationCertificateUpdate, 
    PersonnelQualificationCertificateDel, PersonnelQualificationCertificateQuery
)
from app.services.system.personnel import PersonnelService, PersonnelQualificationCertificateService


router = APIRouter()


# 人员信息相关接口
@router.post('/create', description="新增人员信息")
async def create_personnel(params: PersonnelIn):
    data = await PersonnelService.create(params)
    return await HttpResponse.success(data)


@router.post('/update', description="更新人员信息")
async def update_personnel(params: PersonnelUpdate):
    await PersonnelService.update(params)
    return await HttpResponse.success()


@router.post('/delete', description="删除人员信息")
async def delete_personnel(params: PersonnelDel):
    await PersonnelService.delete(params)
    return await HttpResponse.success()


@router.post('/list', description="获取人员信息列表")
async def get_personnel_list(params: PersonnelQuery):
    data = await PersonnelService.get_list(params)
    return await HttpResponse.success(data)


@router.post('/get', description="根据ID获取人员信息")
async def get_personnel_by_id(id: int):
    data = await PersonnelService.get_by_id(id)
    return await HttpResponse.success(data)


# 员工资质证书相关接口
@router.post('/qualification/create', description="新增员工资质证书")
async def create_personnel_qualification(params: PersonnelQualificationCertificateIn):
    data = await PersonnelQualificationCertificateService.create(params)
    return await HttpResponse.success(data)


@router.post('/qualification/update', description="更新员工资质证书")
async def update_personnel_qualification(params: PersonnelQualificationCertificateUpdate):
    await PersonnelQualificationCertificateService.update(params)
    return await HttpResponse.success()


@router.post('/qualification/delete', description="删除员工资质证书")
async def delete_personnel_qualification(params: PersonnelQualificationCertificateDel):
    await PersonnelQualificationCertificateService.delete(params)
    return await HttpResponse.success()


@router.post('/qualification/list', description="获取员工资质证书列表")
async def get_personnel_qualification_list(params: PersonnelQualificationCertificateQuery):
    data = await PersonnelQualificationCertificateService.get_list(params)
    return await HttpResponse.success(data)


@router.post('/qualification/get', description="根据ID获取员工资质证书")
async def get_personnel_qualification_by_id(id: int):
    data = await PersonnelQualificationCertificateService.get_by_id(id)
    return await HttpResponse.success(data)

File: frontend/src/api/useSystemApi/personnel.ts
import request from '/@/utils/request';

/**
 * 人员管理接口
 * @method getList 获取人员列表
 * @method getById 根据ID获取人员信息
 * @method create 新增人员
 * @method update 更新人员
 * @method delete 删除人员
 */
export function usePersonnelApi() {
  return {
    // 获取人员列表
    getList: (data?: object) => {
      return request({
        url: '/personnel/list',
        method: 'POST',
        data,
      });
    },
    // 根据ID获取人员信息
    getById: (id: number) => {
      return request({
        url: '/personnel/get',
        method: 'POST',
        data: {id},
      });
    },
    // 新增人员
    create: (data?: object) => {
      return request({
        url: '/personnel/create',
        method: 'POST',
        data,
      });
    },
    // 更新人员
    update: (data?: object) => {
      return request({
        url: '/personnel/update',
        method: 'POST',
        data,
      });
    },
    // 删除人员
    delete: (id: number) => {
      return request({
        url: '/personnel/delete',
        method: 'POST',
        data: {id},
      });
    },
  };
}

/**
 * 员工资质证书接口
 * @method getList 获取资质证书列表
 * @method getById 根据ID获取资质证书
 * @method create 新增资质证书
 * @method update 更新资质证书
 * @method delete 删除资质证书
 */
export function useQualificationApi() {
  return {
    // 获取资质证书列表
    getList: (data?: object) => {
      return request({
        url: '/personnel/qualification/list',
        method: 'POST',
        data,
      });
    },
    // 根据ID获取资质证书
    getById: (id: number) => {
      return request({
        url: '/personnel/qualification/get',
        method: 'POST',
        data: {id},
      });
    },
    // 新增资质证书
    create: (data?: object) => {
      return request({
        url: '/personnel/qualification/create',
        method: 'POST',
        data,
      });
    },
    // 更新资质证书
    update: (data?: object) => {
      return request({
        url: '/personnel/qualification/update',
        method: 'POST',
        data,
      });
    },
    // 删除资质证书
    delete: (id: number) => {
      return request({
        url: '/personnel/qualification/delete',
        method: 'POST',
        data: {id},
      });
    },
  };
}
